src/controller/CHIPDeviceController.cpp

```diff
@@ -58,6 +62,20 @@ namespace DeviceController {
 
 using namespace chip::Encoding;
 
+constexpr const char * kDeviceCredentialsKeyPrefix = "DeviceCredentials";
+constexpr const char * kDeviceAddressKeyPrefix     = "DeviceAddress";
+
+// This macro generats a key using node ID an key prefix, and performs the given action
+// on that key.
+#define PERSISTENT_KEY_OP(node, keyPrefix, key, action)                                                                            \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        const size_t len = strlen(keyPrefix);                                                                                      \
+        char key[len + 2 * sizeof(NodeId) + 1];                                                                                    \
```

Comment:
`2*sizeof(NodeId)` because we are assuming that `NodeId` that in hex 2 chars per byte is what our printf will need?  Worth documenting that.  And maybe asserting something about sizeof(NodeId) <= sizeof(uint64_t) to make sure we don't miss bytes?